dango/dex/src/execute.rs

```diff
@@ -546,13 +606,53 @@ fn clear_orders_of_pair(
         .append(Direction::Ask)
         .range(storage, None, None, IterationOrder::Ascending);
 
+    let reserve = RESERVES.may_load(storage, (&base_denom, &quote_denom))?;
+
     // Run the order matching algorithm.
     let MatchingOutcome {
         range,
         volume,
         bids,
         asks,
-    } = match_orders(bid_iter, ask_iter)?;
+    } = match reserve.clone() {
+        None => match_orders(bid_iter, ask_iter)?,
+        Some(reserve) => {
+            // Create the passive liquidity orders if the pair has a pool.
+            let pair = PAIRS.load(storage, (&base_denom, &quote_denom))?;
+            let (passive_bids, passive_asks) =
+                pair.reflect_curve(base_denom.clone(), quote_denom.clone(), &reserve)?;
+
+            // Convert the passive bids and asks iterators to the format required by match_orders.
+            let passive_bids = passive_bids.map(|res| {
+                res.map(|(price, amount)| {
+                    ((price, u64::MAX), Order {
+                        user: dex_addr,
+                        amount,
+                        remaining: amount,
+                        created_at_block_height: current_block_height,
+                    })
```

Comment:
If you use `current_block_height` here then the passive pool is charged the taker fee rate. I think we should charge it the maker rate instead. To do that, just use `0` for the block height.

Reply:
Nvm, this is not correct--you actually charge zero fee for the passive pool in execute.rs